ls1intum/Artemis
# [Bugfix] Lazy Loading Exception

### Checklist


### Motivation and Context
With the changes in develop/multiple-results this bug arose.

### Description
This bug occurs when a tutor is going to asses a modelling exercise of an exam, and tries to display the given exampleSubmissions.
When a tutor is trying to load the Example Submission, at first a 500 Internal Server Error is shown. By the second press of the button, the exampleSubmission is displayed correctly again.
This error message was sent because the server encountered a lazy loading exception, when trying to store the tutorParticipation. With the change in Submission <> Result from 1-1 1-many, the result (of the tutorParticipations' submission) got dereferenced (lazy loading) when the tutorParticipation was saved to the database.
By storing the result before the database save, and then setting it again afterwards, it stays accessible when the tutorParticipation is returned. Now the exception is not thrown anymore, and therefore also no 500 error in the client.

### Steps for Testing


1. Log in to Artemis, create an exam
2. Add exercise groups, and an **modelling exercise**
3. After creating the modelling exercise, edit it and add example submissions, also: evaluate the examplesubmission
4. Participate in the exam with other users
5. After the exam go to exam assessment with a tutor account
6. Go to the assessment of the modelling exercise
7. A green button, for loading the example submissions should appear.
8. Pressing this button triggers the bug
